climpred/bootstrap.py

```diff
@@ -1,14 +1,94 @@
 import inspect
+import multiprocessing
+import warnings
 
+import dask
 import numpy as np
 import xarray as xr
-from tqdm.auto import tqdm
 
 from .checks import has_dims
 from .constants import ALL_COMPARISONS, ALL_METRICS, METRIC_ALIASES
 from .prediction import compute_hindcast, compute_perfect_model, compute_persistence
 from .stats import dpp, varweighted_mean_period
-from .utils import assign_attrs, get_comparison_class, get_metric_class
+from .utils import (
+    _ensure_loaded,
+    _transpose_and_rechunk_to,
+    assign_attrs,
+    get_comparison_class,
+    get_metric_class,
+)
+
+ncpu = multiprocessing.cpu_count()
+
+
+def chunking_parallel_performance_increase_likely(ds, dim):
+    """Check whether chunking might make sense.
+
+    Criteria for potential performance increase:
```

Comment:
can you confirm that this leads to a speed on your machines?

Reply:
I'll post a response to this in the main comments.